packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +323,82 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	it("full initialization of data object should not happen by default", async () => {
+		const TestDataObjectType1 = "@fluid-example/test-dataStore1";
+		const dataStoreFactory1 = new DataObjectFactory(
+			TestDataObjectType1,
+			TestDataObject1,
+			[],
+			[],
+		);
+		const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+			[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+		]);
+		const containerRuntimeFactoryWithDefaultDataStore =
+			getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+		const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[],
+		);
+
+		// Create a container for the first client.
+		const container1 = await provider.createContainer(runtimeFactory);
+		assert.doesNotThrow(
+			async () => requestFluidObject<TestDataObject1>(container1, "/"),
+			"Initial creation of container and data store should succeed.",
+		);
+
+		// Create a summarizer for the container and do a summary shouldn't throw.
+		const createSummarizerResult = await createSummarizerFromFactory(
+			provider,
+			container1,
+			dataStoreFactory1,
+			undefined,
+			containerRuntimeFactoryWithDefaultDataStore,
+			registryStoreEntries,
+		);
+		const summarizer = createSummarizerResult.summarizer;
+		assert.doesNotThrow(async () => {
+			await summarizeNow(summarizer, "test");
+		});
+
+		// In summarizer, load the data store should fail.
+		const requestHeader = {
+			[LoaderHeader.clientDetails]: {
+				capabilities: { interactive: false },
+				type: "summarizer",
+			},
+			[DriverHeader.summarizingClient]: true,
+		};
+		const summarizerContainer = await provider.loadContainer(
+			runtimeFactory,
+			undefined,
+			requestHeader,
+		);
+
+		const expectedErrors = [
+			{
+				eventName: "Error",
+				error: "Non interactive/summarizer client's data object should not be initialized",
+			},
+		];
+		itExpects(
+			"This validates the summarizer client doesn't fully initialize the data object.",
+			expectedErrors,
+			async () => requestFluidObject<TestDataObject1>(summarizerContainer, "/"),
+		);
+
+		// Load second container, load the data store again should success.
```

Comment:
Maybe include in the comment that a second container is created so that it also goes throw the existing flow and calls `initializingFromExisting` which the first container does not.